### PR TITLE
Recode the project in Scala so that it is always compatible with all version of Spark ML engine

### PR DESCRIPTION
**Pull request by valdo404**
Project needs to be updated in order to work with spark 2.2

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -73,20 +73,74 @@
 		<dependency>
 			<groupId>org.apache.spark</groupId>
 			<artifactId>spark-sql_2.11</artifactId>
-			<version>[2.0.0, 2.0.2]</version>
+			<version>[2.1.0, 2.2.99]</version>
```
Comment (kholodilov) [inline pom.xml]: Dependency version specified this way causes Maven to download poms for all possible versions, including snapshots, when building project. For me it took about 30 minutes and still haven't finished. I think it's better to specify minimal compatible version here.

Comment (valdo404) [reply pom.xml]: Will adapt the code.

Comment (kholodilov) [reply pom.xml]: BTW with small workaround I made the transformer compatible with Spark 2.0. See https://github.com/kycml/jpmml-evaluator-spark/commit/0eda0f49faabfdbe41aec913a888460ecc0154c8

Comment (valdo404) [reply pom.xml]: What do you prefer: 
- having a full scala implementation which makes integration with spark easier to read and to test ? (I added bunch of unit tests btw)
- or staying on full java implem ?

let me know.

Comment (valdo404) [reply pom.xml]: @vruusmann let me know your thoughts about this

**Pull request closed without merging**